engine/gamesys/src/gamesys/resources/res_sound_data.cpp

```
@@ -246,7 +247,13 @@ namespace dmGameSystem
 
     dmResource::Result ResSoundDataCreate(const dmResource::ResourceCreateParams* params)
     {
-        dmSound::SoundDataType type = TryToGetTypeFromBuffer((char*)params->m_Buffer, params->m_BufferSize, dmSound::SOUND_DATA_TYPE_WAV);
+        dmSound::SoundDataType type;
+        bool type_result = TryToGetTypeFromBuffer((char*)params->m_Buffer, params->m_BufferSize, &type);
+        if (!type_result)
+        {
+            dmLogError("Failed to detect sound type for: %s", params->m_Filename);
+            return dmResource::RESULT_INVALID_DATA;
```

Fail early if the data is broken